sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/consumer_client.hpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// cspell: word myeventhub
+
+#pragma once
+#include "eventhub_constants.hpp"
+#include "models/consumer_client_models.hpp"
+#include "models/management_models.hpp"
+#include "partition_client.hpp"
+
+#include <azure/core/amqp.hpp>
+#include <azure/core/context.hpp>
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/core/diagnostics/logger.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/diagnostics/log.hpp>
+namespace Azure { namespace Messaging { namespace EventHubs {
+  /**@brief Contains options for the ConsumerClient creation
+   */
+  struct ConsumerClientOptions
+  {
+    /**@brief ApplicationID is used as the identifier when setting the User-Agent property.
+     */
+    std::string ApplicationID = "";
+
+    /**@brief  RetryOptions controls how often operations are retried from this client and any
+     * Receivers and Senders created from this client.
+     */
+    Azure::Core::Http::Policies::RetryOptions RetryOptions{};
+
+    /**@brief  Message sender options.
+     */
+    Azure::Core::Amqp::_internal::MessageReceiverOptions ReceiverOptions{};
+  };
+
+  /**@brief Contains credentials for the ConsumerClient creation
+   */
+  struct ConsumerClientCreds
+  {
+    /// The connection string for the Event Hubs namespace
+    std::string ConnectionString;
+
+    /// the Event Hubs namespace name (ex: myeventhub.servicebus.windows.net)
+    std::string HostName;
+
+    /// The name of the Event Hub
+    std::string EventHub;
+
+    /// The name of the consumer group
+    std::string ConsumerGroup;
+
+    /// Credentials to be used to authenticate the client.
+    std::shared_ptr<Core::Credentials::TokenCredential> Credential{};
+
+    /// The URL to the Event Hubs namespace
+    std::string HostUrl{};
+  };
+
+  /**@brief GetEventHubPropertiesOptions contains optional parameters for the GetEventHubProperties
+   * function
+   */
+  struct GetEventHubPropertiesOptions
+  {
+    // For future expansion
+  };
+
+  /**@brief GetPartitionPropertiesOptions contains optional parameters for the
+   * GetPartitionProperties function
+   */
+  struct GetPartitionPropertiesOptions
+  {
+    // For future expansion
+  };
+
+  /**
+   * @brief The ConsumerClient class is a high level class used to consume events from an Event Hub.
+   *
+   * @details The ConsumerClient class uses a #Azure::Messaging::EventHubs::PartitionClient to
+   * receive events from a specific partition of an Event Hub. The
+   * #Azure::Messaging::EventHubs::PartitionClient is created by the #ConsumerClient and is
+   * available via the NewPartitionClient method. The ConsumerClient is also responsible for
+   * managing the connection to the Event Hub and will reconnect as necessary.
+   */
+  class ConsumerClient {
```

Comment:
```suggestion
  class ConsumerClient final {
```

Alternative - if this class is meant for deriving, add public virtual destructor or protected non-virtual destructor